suite-native/device-authentication/src/deviceAuthenticationSlice.ts

```diff
@@ -0,0 +1,53 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+import { UI } from '@trezor/connect';
+
+import { isPinButtonRequestCode } from './utils';
+
+type DeviceAuthenticationState = {
+    hasDeviceRequestedPin: boolean;
+};
+
+type DeviceAuthenticationRootState = {
+    deviceAuthentication: DeviceAuthenticationState;
+};
+
+const deviceAuthenticationInitialState: DeviceAuthenticationState = {
+    hasDeviceRequestedPin: false,
+};
+
+export const deviceAuthenticationSlice = createSlice({
+    name: 'deviceAuthentication',
+    initialState: deviceAuthenticationInitialState,
+    reducers: {},
+    extraReducers: builder => {
+        builder
+            .addMatcher(
```

Comment:
I think there is no need for using `addMatcher`, you could use `addCase` instead:

```suggestion
            .addCase(UI.REQUEST_PIN, ...
```

Please change it for the other actions as well.

Reply:
Good catch https://github.com/trezor/trezor-suite/pull/12889/commits/74554a00bc14a59b3ef8a21be79a5b4910d3490b